hangman.py

```diff
@@ -0,0 +1,223 @@
+    '''
+
+    guesses_remaining = 6
+    warnings_remaining = 3
+    letters_guessed = []
+    vowels = "aeiou"
+
+    print('Welcome to the game Hangman!')
+    print(f'I am thinking of a word that is {len(secret_word)} letters long.')
+    print(f'You have {warnings_remaining} warnings left')
+    while (guesses_remaining > 0) and not (is_word_guessed(secret_word, letters_guessed)):
+        print('-------------')
+        print(f'You have {guesses_remaining} guesses left.')
+        print(f'Available letters: {get_available_letters(letters_guessed)}')
+
+        let=input("Please guess a letter:").lower()
+
+        #Cheking for symbols quantity
+        if len(let) > 1:
+            print("Your input has more then one symbol")
+            continue
+
+        #Add hints, output all the aceptable words
+        if let == "*":
+            show_possible_matches(get_guessed_word(secret_word, letters_guessed))
+            continue
+
+
+        #Checking for unwanted symbols
+        if not let.isalpha():
+            if warnings_remaining == 0:
+                guesses_remaining -= 1
+            else:
+                warnings_remaining -= 1
+            print("Oops! That is not a valid letter. You have", warnings_remaining, "warnings left:", end="")
+            continue
+
+        letters_guessed.append(let)
+
+        #Cheking for repeating symbols
+        if letters_guessed[-1] in letters_guessed[0:-1]:
+            if warnings_remaining == 0:
+                guesses_remaining -= 1
+            else:
+                warnings_remaining -= 1
+            print("Oops! You've already guessed that letter. You have", warnings_remaining, "warnings left:", end="")
+
+        #Cheking for usuall mistake
+        elif not (letters_guessed[-1] in secret_word):
+            if letters_guessed[-1] in vowels:
+                guesses_remaining -= 2
+            else:
+                guesses_remaining -= 1
+            print("Oops! That letter is not in my word:", end="")
+
+        #Cheking for right answer
+        elif letters_guessed[-1] in secret_word:
+            print("Good guess:", end="")
+
+        print(get_guessed_word(secret_word, letters_guessed))
+
+    if is_word_guessed(secret_word, letters_guessed):
+        print("Congratulations, you won!")
+        print("Your total score for this game is:", guesses_remaining * len(secret_word))
+    else:
+        print("Sorry, you ran out of guesses.")
+        print("The word was else\nIt's", secret_word)
+
+def match_with_gaps(my_word, other_word):
+    '''
+    my_word: string with _ characters, current guess of secret word
+    other_word: string, regular English word
+    returns: boolean, True if all the actual letters of my_word match the
+        corresponding letters of other_word, or the letter is the special symbol
+        _ , and my_word and other_word are of the same length;
+        False otherwise:
+    '''
+
+    emph = "_"
```

Comment:
I believe it makes sense to declare a constant for that

Reply:
+